### ISSUE TITLE
Multiple consoles: support for running scripts in separate process

### ISSUE THREAD

**Issue by jthomasmock** [posit-dev/positron#3173]
We have integrated testing and build processes into alternate panes or background processes as mentioned in #1303 

However, we don't fully support running arbitrary user code in separate processes but the same runtime, i.e., Background Jobs from RStudio. Given that we only have one console, it feels a bit awkward to need to drop into the terminal and lose all of the default behavior to execute long-running scripts in the background.

These may not be as necessary given Python apps/APIs are typically run via the terminal and the Shiny extension runs R/Python via the terminal but it does feel like a gap compared to RStudio.

I also saw this recently of a VS Code R user asking for the `jobs` R package to work with VS Code: https://twitter.com/strengejacke/status/1791224380602622355

**Comment by rmflight** [comment]
Given that multiple consoles / terminals is one of the "killer features" of VSCode, this feels like a "must have" eventually. Glad to see it's on the radar, whether it's for having "background jobs" or just "multiple consoles".

**Comment by abduazizR** [comment]
Not sure if I can comment here, but just want to confirm that these features are not yet implemented but they're in the developers' plans, right?

**Comment by juliasilge** [comment]
Yes, that's right @abduazizR! This feature is not slated for immediate work but it is helpful to know how important it is for our users so adding a 👍 or similar is most welcome.

**Comment by Aariq** [comment]
Just wanted to chime in and say that I'd love for `targets::tar_make(as_job=TRUE)` to "just work" in Positron.  So having some kind of ability to run code in a separate session would be great!

**Comment by seeM** [comment]
Is this related to Jupyter [kernel subshells](https://jupyter-client.readthedocs.io/en/latest/messaging.html#kernel-subshells)?

**Comment by juliasilge** [comment]
For folks who are interested in this feature, please check out this announcement from @dhruvisompura about how to get started with an experimental version of multiple console sessions: <https://github.com/posit-dev/positron/discussions/6712>

**Comment by juliasilge** [comment]
I'm moving this to Ready for Verification, on the basis of our current level of support for multiple consoles. See #6712 for how to opt in, https://github.com/posit-dev/positron/issues/5990 to track further work for our RC UI plans, and the [multisession console label](https://github.com/posit-dev/positron/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22theme%3A%20multisession%20console%22) for current known problems.

**Comment by testlabauto** [comment]
### Verified Fixed
    Positron Version(s) : main
    OS Version          : ubuntu, windows

### Test scenario(s) 

Marking as verified based on the current level of automated sessions testing we have in place.

Link(s) to TestRail test cases run or created:
